x-pack/plugins/infra/public/pages/metrics/hosts/hosts_content.tsx

```diff
@@ -6,82 +6,102 @@
  */
 
 import type { Query, TimeRange } from '@kbn/es-query';
-import { useKibana } from '@kbn/kibana-react-plugin/public';
 import { i18n } from '@kbn/i18n';
 import React, { useState, useCallback } from 'react';
 import { SearchBar } from '@kbn/unified-search-plugin/public';
+import { EuiSpacer } from '@elastic/eui';
+import { NoData } from '../../../components/empty_states';
 import { InfraLoadingPanel } from '../../../components/loading';
 import { useMetricsDataViewContext } from './hooks/use_data_view';
 import { HostsTable } from './components/hosts_table';
-import { InfraClientStartDeps } from '../../../types';
 import { useSourceContext } from '../../../containers/metrics_source';
+import { useSnapshot } from '../inventory_view/hooks/use_snaphot';
+import { SnapshotMetricType } from '../../../../common/inventory_models/types';
 
 export const HostsContent: React.FunctionComponent = () => {
-  const {
-    services: { data },
-  } = useKibana<InfraClientStartDeps>();
-  const { source } = useSourceContext();
+  const { source, sourceId } = useSourceContext();
   const [dateRange, setDateRange] = useState<TimeRange>({ from: 'now-15m', to: 'now' });
   const [query, setQuery] = useState<Query>({ query: '', language: 'kuery' });
   const { metricsDataView, hasFailedCreatingDataView, hasFailedFetchingDataView } =
     useMetricsDataViewContext();
   // needed to refresh the lens table when filters havent changed
-  const [searchSessionId, setSearchSessionId] = useState(data.search.session.start());
-  const [isLensLoading, setIsLensLoading] = useState(false);
 
   const onQuerySubmit = useCallback(
     (payload: { dateRange: TimeRange; query?: Query }) => {
       setDateRange(payload.dateRange);
       if (payload.query) {
         setQuery(payload.query);
       }
-      setIsLensLoading(true);
-      setSearchSessionId(data.search.session.start());
     },
-    [setDateRange, setQuery, data.search.session]
+    [setDateRange, setQuery]
   );
 
-  const onLoading = useCallback(
-    (isLoading: boolean) => {
-      if (isLensLoading) {
-        setIsLensLoading(isLoading);
-      }
-    },
-    [setIsLensLoading, isLensLoading]
+  const hostMetrics: Array<{ type: SnapshotMetricType }> = [
+    { type: 'rx' },
+    { type: 'tx' },
+    { type: 'memory' },
+    { type: 'cpuCores' },
+    { type: 'memoryTotal' },
+  ];
+
+  const { loading, nodes, reload } = useSnapshot(
+    '', // use the unified search query, supported type?
+    hostMetrics,
+    [],
+    'host',
+    sourceId,
+    1665748800000, // currentTime.  need to add support for TimeRange?
+    '',
+    '',
+    true,
+    {
```

Comment:
Regarding the dynamic time range, we discussed with @crespocarlos that it will be better to be added as part of the [filter issue](https://github.com/orgs/elastic/projects/664/views/36) and maybe after we have the [url state](https://github.com/orgs/elastic/projects/664/views/36) implemented so we will have a clear scope of this issue (replacement of the lens table with eui table) and it will unblock the other issues where we need the table to be changed.  What do you think?

Reply:
It makes more sense to address this on the unified search ticket. Everything related to filter is connected to that.